### PR TITLE
[IMP] docker-alpine-samba-dc: Break build args into dot env

### PR DESCRIPTION
**Pull request by tedsalmon**
* Move all the build args out of Dockerfile and into dot env
* Move AD DC initialization into entry point

FYI, the `mkdir` prior to the initial provisioning of the domain exists because any volume mounted post-image creation is going to be empty.

### CODE REVIEW

**Comment by lasley** [comment]
Depends on LasLabs/docker-quality-tools#10

**Comment by tedsalmon** [comment]
Alright, build is 🍏

**Review by lasley — CHANGES_REQUESTED** [review]
What about the ReadMe? You should also add usage sample(s)

**Comment by tedsalmon** [comment]
Updated, @lasley!

File: README.md [hunk 1]
```diff
@@ -12,12 +12,30 @@ This image provides a Samba 4 Domain Controller using an Alpine Linux base.
...
+```
+mkdir ${PWD}/samba
+
+docker run --rm -i -t \
+    -e SAMBA_DC_REALM='corp.example.net' \
+    -e SAMBA_DC_DOMAIN='EXAMPLE' \
+    -e SAMBA_DC_ADMIN_PASSWD='5u3r53cur3!' \ 
+    -e SAMBA_DC_DNS_BACKEND='SAMBA_INTERNAL' \
+    -v ${PWD}/samba:/samba \
+     'samba-dc:1.0'
```
Comment (lasley) [inline README.md]: `laslabs/alpine-samba-dc:latest`

File: README.md [hunk 2]
```diff
@@ -12,12 +12,30 @@ This image provides a Samba 4 Domain Controller using an Alpine Linux base.
 Configuration
 =============
 
-*
+After building the image, you should start it with the correct environment
```
Comment (lasley) [inline README.md]: There's no reason to build. Simply `docker pull laslabs/alpine-samba-dc:latest`

File: README.md [hunk 3]
```diff
@@ -12,12 +12,30 @@ This image provides a Samba 4 Domain Controller using an Alpine Linux base.
...
+    -e SAMBA_DC_DNS_BACKEND='SAMBA_INTERNAL' \
+    -v ${PWD}/samba:/samba \
+     'samba-dc:1.0'
+```
 
 Usage
 =====
-
-*
+After it's configured, you can run the container like so:
```
Comment (lasley) [inline README.md]: Maybe "After the container has been run the first time" or something?

File: README.md [hunk 4]
```diff
@@ -12,12 +12,30 @@ This image provides a Samba 4 Domain Controller using an Alpine Linux base.
...
+```
+docker run --rm -i -t -v ${PWD}/samba:/samba 'samba-dc:1.0'
```
Comment (lasley) [inline README.md]: Why `--rm`?

File: README.md [hunk 5]
```diff
@@ -12,12 +12,30 @@ This image provides a Samba 4 Domain Controller using an Alpine Linux base.
...
+After building the image, you should start it with the correct environment 
+variables for initial configuration:
+
+* Note that we're persisting the samba volume to a local directory named 
+`samba` in your working dir
+
+```
+mkdir ${PWD}/samba
+
+docker run --rm -i -t \
```
Comment (lasley) [inline README.md]: Why `--rm`? Also, why interactive? Isn't this a Daemon?

File: docker-entrypoint.sh [hunk 6]
```diff
@@ -11,6 +11,16 @@ if [ "${1:0:1}" = '-' ]; then
 	set -- $COMMAND "$@"
 fi
 
+# Configure the AD DC
+if [ ! -f /samba/etc/smb.conf ]; then
+    mkdir -p /samba/etc /samba/lib /samba/log
```
Comment (lasley) [inline docker-entrypoint.sh:16]: `echo "${SAMBA_DC_DOMAIN} - Begin Provisioning"`

File: docker-entrypoint.sh [hunk 7]
```diff
@@ -11,6 +11,16 @@ if [ "${1:0:1}" = '-' ]; then
...
+    samba-tool domain provision --domain="${SAMBA_DC_DOMAIN}" \
+        --adminpass="${SAMBA_DC_ADMIN_PASSWD}" \
+        --server-role=dc \
+        --realm="${SAMBA_DC_REALM}" \
+        --dns-backend="${SAMBA_DC_DNS_BACKEND}"
+fi
```
Comment (lasley) [inline docker-entrypoint.sh]: `echo "${SAMBA_DC_DOMAIN} - Provisioned Successfully"`

**Review by lasley — CHANGES_REQUESTED** [review]

File: README.md (hunk 5 above)
Comment (lasley) [reply README.md]: Maybe also `--restart unless-stopped`

**Comment by tedsalmon** [comment]
Updated -- The Docker commands were a carryover of what I was using to test locally, derp. When you approve, I'll squash and merge.

Thanks!

File: README.md [hunk 8]
```diff
@@ -11,13 +11,35 @@ This image provides a Samba 4 Domain Controller using an Alpine Linux base.
 
 Configuration
 =============
+First, Pull the image:
+```
+docker pull laslabs/alpine-samba-dc:latest
+```
 
-*
+Now, start the image with the correct environmentvariables for initial
```
Comment (lasley) [inline README.md]: s/environmentvariables/environment variables

Comment (tedsalmon) [reply README.md]: Odd -- Fixed! :)

**Review by lasley — APPROVED** [review]

**Pull request merged**